File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        st1=str(x)
        st2=""

        if st1[0]=='-':
            
            for i in range(1,len(st1)):
                st2+=st1[len(st1)-i]
            n=-(int(st2))
        else:
            for i in range(1,len(st1)+1):
                st2+=st1[len(st1)-i]
            n=int(st2)
        if n>(-2**31) and n<((2**31)-1):
            return n
        else:
            return 0
        
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int reverse_int=0;
        
        while(x!=0){
            int a=x%10;
            if (reverse_int<(INT_MIN/10)|| reverse_int>(INT_MAX/10)){
                return 0;
            }
            reverse_int=(reverse_int*10)+a;
            x=x/10;

        
        }
        return reverse_int;
        
    }
};